src/main/java/com/microsoft/graph/tasks/LargeFileUploadTask.java

```diff
@@ -159,36 +159,43 @@ public ChunkedUploadProvider(@Nonnull final IUploadSession uploadSession,
+                    new LargeFileUploadRequest<>(this.uploadUrl, this.client, options, buffer, buffRead,
                             this.readSoFar, this.streamSize);
-            final ChunkedUploadResult<UploadType> result = request.upload(this.responseHandler);
+            final LargeFileUploadResponse<UploadType> response = request.upload(this.responseHandler);
             // TODO: upload should return a future, use sendfuture instead and the futures should be chained with completableFuture.then apply
 
-            if (result.uploadCompleted()) {
+            if (response.uploadCompleted()) {
                 if(progressCallback != null) {
                     progressCallback.progress(this.streamSize, this.streamSize);
                 }
-                return completedFuture(result.getItem());
-            } else if (result.chunkCompleted()) {
+                final LargeFileUploadResult<UploadType> result = new LargeFileUploadResult<UploadType>();
+                if (response.getItem() != null) {
+                    result.responseBody = response.getItem();
+                }
+                if (response.getLocation() != null) {
+                    result.location = response.getLocation();
+                }
+                return completedFuture(result);
+            } else if (response.chunkCompleted()) {
                 if(progressCallback != null) {
                     progressCallback.progress(this.readSoFar, this.streamSize);
                 }
-            } else if (result.hasError()) {
-                return failedFuture(result.getError());
+            } else if (response.hasError()) {
+                return failedFuture(response.getError());
             }
 
             this.readSoFar += buffRead;
         }
         return failedFuture(new ClientException("Upload did not complete", null));
     }
-    private CompletableFuture<UploadType> completedFuture(final UploadType result) { // CompletableFuture.completedFuture(result.getItem()); missing on android
-        final CompletableFuture<UploadType> fut = new CompletableFuture<UploadType>();
+    private java.util.concurrent.CompletableFuture<LargeFileUploadResult<UploadType>> completedFuture(final LargeFileUploadResult<UploadType> result) { // CompletableFuture.completedFuture(result.getItem()); missing on android
+        final CompletableFuture<LargeFileUploadResult<UploadType>> fut = new CompletableFuture<LargeFileUploadResult<UploadType>>();
         fut.complete(result);
         return fut;
     }
-    private CompletableFuture<UploadType> failedFuture(ClientException ex) { // CompletableFuture.failedFuture not available on android
-        final CompletableFuture<UploadType> fut = new CompletableFuture<UploadType>();
+    private java.util.concurrent.CompletableFuture<LargeFileUploadResult<UploadType>> failedFuture(ClientException ex) { // CompletableFuture.failedFuture not available on android
```

Comment:
nit: no need for this to be fully qualified. Same with line 192

Reply:
I wanted to be consistent with other methods but this just makes it harder to read with no real benefit, removing everywhere.